src/object.d

```diff
@@ -4956,3 +4956,315 @@ string _d_assert_fail(string comp, A, B)(A a, B b) @nogc @safe nothrow pure
     buffer[n .. n + valB.length] = valB;
     return (() @trusted => cast(string) buffer)();
 }
+
+// These are needed to fake the safety level to `@safe pure nothrow`.
+// Without this code will break as these hooks where previously exported
+// as having this safety level.
+T[] _d_arrayappendT(T)(ref T[] x, scope T[] y) @trusted
+{
+    alias func(T) = T[] function(ref T[] x, scope T[] y) @trusted pure nothrow;
+    auto f = cast(func!T)&_d_arrayappendT_!T;
+    return f(x, y);
+}
+
+T[] _d_arrayappendcTX(T)(ref T[] px, size_t n) @trusted
+{
+    alias func(T) = T[] function(ref T[] px, size_t n) @trusted pure nothrow;
+    auto f = cast(func!T)&_d_arrayappendcTX_!T;
+    return f(px, n);
+}
+
+T[] _d_arrayappendcT(T)(ref T[] px, scope T y) @trusted
+{
+    import core.internal.traits : Unqual;
+    import core.stdc.string : memcpy;
+    auto result = _d_arrayappendcTX(px, 1);
+    memcpy(cast(Unqual!T *)&result[$ - 1], cast(Unqual!T *)&y, T.sizeof);
+    return result;
+}
+
+/**
+ * Run postblit on `t` if it is a struct and needs it.
+ * Or if `t` is a array, run it on the children if they have a postblit.
+ */
+void __doPostblit(T)(auto ref T t)
+{
+    import core.internal.traits : hasElaborateCopyConstructor;
+
+    static if (is(T == struct)) {
+        // run the postblit function incase the struct has one
+        static if (__traits(hasMember, T, "__xpostblit") &&
+                // Bugzilla 14746: Check that it's the exact member of S.
+                __traits(isSame, T, __traits(parent, t.__xpostblit)))
+            t.__xpostblit();
+    }
+    else static if (is(T U : U[]) && hasElaborateCopyConstructor!U)
+    {
+        // only do a postblit if the `U` requires it.
+        foreach (ref el; t)
+            __doPostblit(el);
+    }
+}
+
+T[] _d_arrayappendTTrace(T)(string file, int line, string funcname, ref T[] x, scope T[] y) @trusted pure
+{
+    import rt.profilegc : accumulate;
+
+    string typename = T.stringof;
+    version (tracegc)
+    {
+        printf("_d_arrayappendT type = %.*s x = %llu y = %llu file = '%.*s' line = %d function = '%.*s'\n",
+            cast(int)typename.length, typename.ptr,
+            cast(ulong)x.length,
+            cast(ulong)y.length,
+            file.length, file.ptr,
+            line,
+            funcname.length, funcname.ptr
+            );
+    }
+    accumulate(file, line, funcname, typename, T.sizeof * y.length);
+    return _d_arrayappendT(x, y);
+}
+
+/**
+ * Append y[] to array x[]
+ */
+T[] _d_arrayappendT_(T)(ref T[] x, scope T[] y) @trusted
```

Comment:
It looks like this is only used in one place.  What about embedding it in the caller as a nested function to make it more obvious that this is an implementation detail of that function?

Reply:
Good suggestion, will do.